src/main/kotlin/app/filesystem/DirectoryProvider.kt

```diff
@@ -26,7 +27,7 @@ class DirectoryProvider(private val appName: String) {
     }
 
     // create a directory to store the user's application projects/documents
-    fun getUserDataDirectory(appendedPath: String = "", createIfNotExists: Boolean = true) : String {
+    override fun getUserDataDirectory(appendedPath: String, createIfNotExists: Boolean) : String {
         // create the directory if it does not exist
         var pathString = System.getProperty("user.home") + separator + appName
```

Comment:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html

Reply:
@sarabiaj Do you want us to switch to using `joinToString` throughout DirectoryProvider?

Reply:
yeah lol